src/scaffoldfitter/fitterstepfit.py

```diff
@@ -186,142 +188,108 @@ def run(self, modelFileNameStem=None):
             if self._edgeDiscontinuityPenaltyWeight > 0.0:
-                edgeDiscontinuityPenaltyObjective = self.createEdgeDiscontinuityPenaltyObjectiveField()
-                result = optimisation.addObjectiveField(edgeDiscontinuityPenaltyObjective)
-                assert result == RESULT_OK, "Fit Geometry:  Could not add edge discontinuity penalty objective field"
+                print("WARNING! Edge discontinuity penalty is not supported by NEWTON solver - skipping")
+                #edgeDiscontinuityPenaltyObjective = self.createEdgeDiscontinuityPenaltyObjectiveField()
+                #result = optimisation.addObjectiveField(edgeDiscontinuityPenaltyObjective)
+                #assert result == RESULT_OK, "Fit Geometry:  Could not add edge discontinuity penalty objective field"
 
         fieldcache = fieldmodule.createFieldcache()
+        objectiveFormat = "{:12e}"
         for iter in range(self._numberOfIterations):
+            iterName = str(iter + 1)
             if self.getDiagnosticLevel() > 0:
-                print("-------- Iteration", iter + 1)
+                print("-------- Iteration " + iterName)
             if self.getDiagnosticLevel() > 0:
-                for d in range(2):
-                    if dataProjectionObjective[d]:
-                        result, objective = dataProjectionObjective[d].evaluateReal(fieldcache, dataProjectionObjectiveComponentsCount[d])
-                        print("    " + str(d + 1) + "-D data projection objective", objective)
-                if markerObjectiveField:
-                    result, objective = markerObjectiveField.evaluateReal(fieldcache, markerObjectiveField.getNumberOfComponents())
-                    print("    marker objective", objective)
+                result, objective = dataObjective.evaluateReal(fieldcache, 1)
+                print("    Data objective", objectiveFormat.format(objective))
                 if deformationPenaltyObjective:
                     result, objective = deformationPenaltyObjective.evaluateReal(fieldcache, deformationPenaltyObjective.getNumberOfComponents())
-                    print("    deformation penalty objective", objective)
+                    print("    Deformation penalty objective", objectiveFormat.format(objective))
             result = optimisation.optimise()
             if self.getDiagnosticLevel() > 1:
                 solutionReport = optimisation.getSolutionReport()
                 print(solutionReport)
             assert result == RESULT_OK, "Fit Geometry:  Optimisation failed with result " + str(result)
-            self._fitter.calculateDataProjections(self)
             if modelFileNameStem:
-                self._fitter.writeModel(modelFileNameStem + "_fit" + str(iter + 1) + ".exf")
+                self._fitter.writeModel(modelFileNameStem + "_fit" + iterName + ".exf")
+            self._fitter.calculateDataProjections(self)
         if self.getDiagnosticLevel() > 0:
```

Comment:
Can we remove this `if` statement and just add the `print("--")` to the next `if` statement?